Server/ConcordiaCurriculumManager/Controllers/GroupController.cs

```diff
@@ -0,0 +1,68 @@
+﻿using ConcordiaCurriculumManager.DTO;
+using ConcordiaCurriculumManager.Models.Users;
+using ConcordiaCurriculumManager.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace ConcordiaCurriculumManager.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+
```

Comment:
Everything looks good, can you just add an `[Authorize]` attribute to the controller level to require users to be logged in to access any method here. Additionally, at the method level for the CreateGroup, AddUserToGroup, and RemoveUserFromGroup, can you require the admin role (see the other controllers for examples of role-based auth, it's just an attribute to add)